olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -107,48 +106,48 @@ client::CancellationToken VersionedLayerClientImpl::GetPartitions(
   return token;
 }
 
-olp::client::CancellationToken VersionedLayerClientImpl::GetData(
-    DataRequest data_request, Callback callback) const {
-  auto fetch_option = data_request.GetFetchOption();
-  if (fetch_option == CacheWithUpdate) {
-    auto cache_token = AddGetDataTask(data_request.WithFetchOption(CacheOnly),
-                                      std::move(callback));
-    auto online_token =
-        AddGetDataTask(data_request.WithFetchOption(OnlineIfNotFound), nullptr);
-    return client::CancellationToken([cache_token, online_token]() {
-      cache_token.cancel();
-      online_token.cancel();
-    });
-  } else {
-    return AddGetDataTask(data_request, std::move(callback));
-  }
-}
-
-client::CancellationToken VersionedLayerClientImpl::AddGetDataTask(
+client::CancellationToken VersionedLayerClientImpl::GetData(
```

Comment:
Why you moved the `AddGetDataTask` to the lambda? For me previous approach with separate method was better than creating the lambda - easier to write unit tests.

Reply:
It's easier to take FetchOptions block to another place and leave a lambda inside, in this scenario 1 method per 1 API call.